### PR TITLE
[Flow EVM] use transfer from/to a fixed native token bridge address for deposit/withdraw calls

### PR DESCRIPTION
**Pull request by ramtinms**
This PR 
- changes the deposit to adjust a balance on a fixed pre-allocated address and then do a simple transfer call from the bridge address to the target address. 
- changes the withdraw call to be a simple transfer call to the bridge address followed by a balance subtraction for the bridge account. 

But why?
1. This is the first PR towards #5468, using simple transfer makes it possible to deposit to any address (even smart contracts). 

2. Not using a fixed pre-allocated address, means all the tools should be updated to expect a COA account balance changes after bridging and some legacy tools might not know how to deal with these anomalies, using a fixed address, manage the anomaly to a single EOA (bridge account). 

3. This would allow later expansion of COA smart contracts to introduce TokenDeposit events like other smart contract wallets.

### CODE REVIEW

File: fvm/evm/emulator/emulator.go [hunk 1]
```diff
@@ -381,12 +370,16 @@ func (proc *procedure) runDirect(
...
 	txIndex uint,
-	txType uint8,
 ) (*types.Result, error) {
 	// set the nonce for the message (needed for some opeartions like deployment)
 	msg.Nonce = proc.state.GetNonce(msg.From)
 	proc.evm.TxContext.Origin = msg.From
-	return proc.run(msg, txHash, txIndex, types.DirectCallTxType)
+	res, err := proc.run(msg, txHash, txIndex, types.DirectCallTxType)
+	if err != nil {
+		return res, err
```
Comment (m-Peter) [inline fvm/evm/emulator/emulator.go:379]: Do we want to return `res` here? Shouldn't it be `return nil, err` ? Or is there a reason for it?

File: fvm/evm/emulator/emulator.go [hunk 2]
```diff
@@ -155,8 +155,8 @@ func (bl *BlockView) RunTransaction(
 	if err != nil {
 		return nil, err
 	}
-	res.TxHash = txHash
```
Comment (m-Peter) [inline fvm/evm/emulator/emulator.go:158]: Is the `TxHash` being set in another way now?

**Review by m-Peter — APPROVED** [review]
Looks nice :ok_hand:

File: fvm/evm/handler/handler.go [hunk 3]
```diff
@@ -471,6 +471,7 @@ func (a *Account) deposit(v *types.FLOWTokenVault) error {
 	bridgeAccount := a.fch.AccountByAddress(bridge, false)
 
 	call := types.NewDepositCall(
+		a.fch.addressAllocator.NativeTokenBridgeAddress(),
```
Comment (sideninja) [inline fvm/evm/handler/handler.go:474]:
```suggestion
    bridge,
```

File: fvm/evm/emulator/emulator.go [hunk 4]
```diff
@@ -200,64 +200,53 @@ func (proc *procedure) commit() error {
...
-		TxType:      types.DirectCallTxType,
-		TxHash:      txHash,
-	}
+	bridge := call.From.ToCommon()
 
-	// create account if not exist
-	if !proc.state.Exist(addr) {
-		proc.state.CreateAccount(addr)
+	// create bridge account if not exist
+	if !proc.state.Exist(bridge) {
```
Comment (sideninja) [inline fvm/evm/emulator/emulator.go:209]: just wondering why we don't create these "special" accounts as part of evm setup?

File: fvm/evm/emulator/emulator.go (hunk 2 above)
Comment (sideninja) [reply fvm/evm/emulator/emulator.go:158]: it's assigned in the procedure itself

File: fvm/evm/types/call.go [hunk 5]
```diff
@@ -39,7 +43,7 @@ const (
 // Note that eventhough we don't check the nonce, it impacts
 // hash calculation and also impacts the address of resulting contract
 // when deployed through direct calls.
-// Users don't have the worry about the nonce, and the emulator would set
+// Users don't have the worry about the nonce, and the handler sets
```
Comment (sideninja) [inline fvm/evm/types/call.go:46]:
```suggestion
// Users don't have the worry about the nonce, handler sets
```

**Review by sideninja — APPROVED** [review]
very nice

**Review by janezpodhostnik — APPROVED** [review]
LGTM, except the comment from @m-Peter https://github.com/onflow/flow-go/pull/5469/files#r1508162324

File: fvm/evm/emulator/emulator.go (hunk 2 above)
Comment (ramtinms) [reply fvm/evm/emulator/emulator.go:158]: Yeah its set inside the run method, it was a duplicated logic

File: fvm/evm/emulator/emulator.go (hunk 4 above)
Comment (ramtinms) [reply fvm/evm/emulator/emulator.go:209]: hmm, we could, I was more leaning towards lazy creation, but we can revisit this in a future PR for all of the special accounts.

File: fvm/evm/types/call.go (hunk 5 above)
Comment (ramtinms) [reply fvm/evm/types/call.go]: thanks

File: fvm/evm/handler/handler.go (hunk 3 above)
Comment (ramtinms) [reply fvm/evm/handler/handler.go]: nice, fixed it.

File: fvm/evm/emulator/emulator.go (hunk 1 above)
Comment (ramtinms) [reply fvm/evm/emulator/emulator.go]: fixed it to return res

**Pull request merged**